pkg/limayaml/validate.go

```diff
@@ -290,15 +293,21 @@ func validateNetwork(y LimaYAML, warn bool) error {
 			if config.Check(nw.Lima) != nil {
 				return fmt.Errorf("field `%s.lima` references network %q which is not defined in networks.yaml", field, nw.Lima)
 			}
+		} else if nw.Socket != "" {
+			if fi, err := os.Stat(nw.Socket); err != nil && !errors.Is(err, os.ErrNotExist) {
+				return err
+			} else if err == nil && fi.Mode()&os.ModeSocket == 0 {
+				return fmt.Errorf("field `%s.socket` %q points to a non-socket file", field, nw.Socket)
+			}
```

Comment:
This branch needs the corresponding tests for `VNLDeprecated != ""` and `SwitchPortDeprecated != 0` from the Lima branch above ("%.socket and %.vml are mutually exclusive", and "%.switchPort cannot be used with %.socket")

Reply:
Fixed